tests/test_nlg.py

```diff
@@ -79,3 +81,45 @@ def test_nlg_schema_validation_empty_buttons():
 def test_nlg_schema_validation_empty_image():
     content = {"text": "Hey there!", "image": None}
     assert CallbackNaturalLanguageGenerator.validate_response(content)
+
+
+@pytest.mark.parametrize("filled_slots", [
+    {"tag_w_underscore": "a"},
+    {"tag with space": "bacon", "tag_w_int_val": 123},
+    {"tag.with.dot": "chocolate", "tag_2": "b"},
+    {"tag-w-dash": "apple pie", "tag.with.float.val": 1.3},
+    {"tag-w-$": "banana"},
+    {"tag-w-@": "one", "tagCamelCase": "two", "tag-w-*": "three"},
+])
+def test_nlg_fill_template_text(filled_slots):
+    template_text = ", ".join(["{" + t + "}" for t in filled_slots.keys()])
```

Comment:
You can also use parametrize a bit differently so you don't have to do the strings joins (https://docs.pytest.org/en/latest/parametrize.html):

```python
@pytest.mark.parametrize("slot_name,slot_value", [("tag_w_underscore", "a"),]
def test_nlg_fill_template_text(slot_name, slot_value): 
   pass 
```